### DISCUSSION TITLE
Waiting for i2c slave device to be powered on

### DISCUSSION THREAD

**Discussion by uGinn-R** [micropython/micropython#12409] [ESP8266]
I'm writing the script for esp8266 that is acting as i2c master (using machine.i2c) for communicating with another i2c device. Is there a correct way to wait in a loop for slave device to be powered on so i can start read\write operations after that? Any suggestions are welcome!

**Comment by robert-hh** [comment]
You can run i2c.scan() and wait for the device's address to show up in the result list.

**Reply by uGinn-R** [reply]
Thanks for your reply! I tried this method, but for some reason it responds with very big delay (like 10 seconds or so), and sometimes even hangs... Here is example:
```
while not DEVICE_ADDRESS in i2c.scan():
     print("waiting...")
     time.sleep(1)
```

**Reply by robert-hh** [reply]
That happens when the bus is not pulled up. You should add pull-up resistors driven by Vcc of the ESP8266.

**Reply by uGinn-R** [reply]
Thanks! I'll try to add pull-up to the scl/sda and report the results :) Cheers!
UPD. Should i add the pull-up resistors if the bus is already pulled up on the slave side? Master and slave are both 3v3 powered but the sources are different.
